### PR TITLE
[flutter_tools] Precache after channel switch

### LINKED ISSUE #44118: Cached artifacts are not updated after switching channels
See the following log. This was on an upgrade of the `master` channel on Windows 10. Notice that there are two attempts to download the Dart SDK, with different engine hashes. 
```
C:\git\dart_console>flutter upgrade
Checking Dart SDK version...
Downloading Dart SDK from Flutter engine b875c7a5ff5680d3c734e3b94ba707b4f3a9527a...
Unzipping Dart SDK...
Building flutter tool...
Running pub upgrade...
Upgrading Flutter from C:\git\flutter...
From github.com:flutter/flutter
   48ce60824..3422540bc  master                  -> origin/master
 * [new branch]          Hixie-patch-2           -> origin/Hixie-patch-2
 * [new branch]          Hixie-patch-3           -> origin/Hixie-patch-3
 * [new branch]          Hixie-patch-4           -> origin/Hixie-patch-4
 + 2d2a1ffec...e70236e36 beta                    -> origin/beta  (forced update)
   3cf88fed6..1946fc4da  dev                     -> origin/dev
 * [new branch]          hot_restart_lowercase_r -> origin/hot_restart_lowercase_r
 * [new branch]          revert-40607-animationOverview -> origin/revert-40607-animationOverview
 * [new branch]          revert-40710-flutterGold -> origin/revert-40710-flutterGold
 * [new branch]          revert-41415-exposeimage -> origin/revert-41415-exposeimage
 * [new branch]          revert-41780-fix_build  -> origin/revert-41780-fix_build
 * [new branch]          sjindel.elf             -> origin/sjindel.elf
   2d2a1ffec..68587a091  stable                  -> origin/stable
   1ced75cc2..eb6877bf9  v1.9.1-hotfixes         -> origin/v1.9.1-hotfixes
 * [new tag]             v1.10.14                -> v1.10.14
 * [new tag]             v1.10.7                 -> v1.10.7
 * [new tag]             v1.9.1+hotfix.6         -> v1.9.1+hotfix.6
 * [new tag]             v1.10.10                -> v1.10.10
 * [new tag]             v1.10.11                -> v1.10.11
 * [new tag]             v1.10.12                -> v1.10.12
 * [new tag]             v1.10.13                -> v1.10.13
 * [new tag]             v1.10.6                 -> v1.10.6
 * [new tag]             v1.10.8                 -> v1.10.8
 * [new tag]             v1.10.9                 -> v1.10.9
 * [new tag]             v1.9.1+hotfix.3         -> v1.9.1+hotfix.3
 * [new tag]             v1.9.1+hotfix.4         -> v1.9.1+hotfix.4
 * [new tag]             v1.9.1+hotfix.5         -> v1.9.1+hotfix.5
Checking out files:  93% (1154/1229)
Checking out files:  94% (1156/1229)
Checking out files:  95% (1168/1229)
Checking out files:  96% (1180/1229)
Checking out files:  97% (1193/1229)
Checking out files:  98% (1205/1229)
Checking out files:  99% (1217/1229)
Checking out files: 100% (1229/1229)
Checking out files: 100% (1229/1229), done.
Updating 3cf88fed6..3422540bc
 .../AppIcon.appiconset/app_icon_1024.png           |  Bin 0 -> 46993 bytes
 .../AppIcon.appiconset/app_icon_128.png            |  Bin 0 -> 3276 bytes
 .../AppIcon.appiconset/app_icon_16.png             |  Bin 0 -> 1429 bytes
 .../AppIcon.appiconset/app_icon_256.png            |  Bin 0 -> 5933 bytes
 .../AppIcon.appiconset/app_icon_32.png             |  Bin 0 -> 1243 bytes
 .../AppIcon.appiconset/app_icon_512.png            |  Bin 0 -> 14800 bytes
 .../AppIcon.appiconset/app_icon_64.png             |  Bin 0 -> 1874 bytes
 .../AppIcon.appiconset/app_icon_1024.png           |  Bin 0 -> 46993 bytes
 .../AppIcon.appiconset/app_icon_128.png            |  Bin 0 -> 3276 bytes
 .../AppIcon.appiconset/app_icon_16.png             |  Bin 0 -> 1429 bytes
 .../AppIcon.appiconset/app_icon_256.png            |  Bin 0 -> 5933 bytes
 .../AppIcon.appiconset/app_icon_32.png             |  Bin 0 -> 1243 bytes
 .../AppIcon.appiconset/app_icon_512.png            |  Bin 0 -> 14800 bytes
 .../AppIcon.appiconset/app_icon_64.png             |  Bin 0 -> 1874 bytes
 .../hermetic}/devices_test.dart                    |    0
 .../hermetic}/install_test.dart                    |    0
 .../permeable}/format_test.dart                    |    0
 .../permeable}/upgrade_test.dart                   |    0
 1200 files changed, 58861 insertions(+), 20479 deletions(-)
Checking Dart SDK version...
Downloading Dart SDK from Flutter engine 8ea19b1c76f69f7b50e4195f41dfb1cb49a9fe66...
Unzipping Dart SDK...
Building flutter tool...
Running pub upgrade...

Upgrading engine...
Downloading Android Maven dependencies...                          15.6s
Downloading android-arm-profile/windows-x64 tools...                0.5s
Downloading android-arm-release/windows-x64 tools...                0.4s
Downloading android-arm64-profile/windows-x64 tools...              0.5s
Downloading android-arm64-release/windows-x64 tools...              0.4s
Downloading android-x64-profile/windows-x64 tools...                0.5s
Downloading android-x64-release/windows-x64 tools...                0.6s
Downloading android-x86 tools...                                    2.0s
Downloading android-x64 tools...                                    1.9s
Downloading android-arm tools...                                    1.0s
Downloading android-arm-profile tools...                            0.5s
Downloading android-arm-release tools...                            0.5s
Downloading android-arm64 tools...                                  0.9s
Downloading android-arm64-profile tools...                          0.6s
Downloading android-arm64-release tools...                          0.5s
Downloading android-x64-profile tools...                            0.6s
Downloading android-x64-release tools...                            0.5s
Downloading android-x86-jit-release tools...                        0.7s
Downloading package sky_engine...                                   0.2s
Downloading common tools...                                         0.9s
Downloading common tools...                                         1.0s
Downloading windows-x64 tools...                                    2.5s

Flutter 1.10.15-pre.387 • channel master • git@github.com:flutter/flutter.git
Framework • revision 3422540bc4 (73 minutes ago) • 2019-11-04 09:09:20 -0800
Engine • revision 8ea19b1c76
Tools • Dart 2.6.0 (build 2.6.0-dev.8.2 bbe2ac28c9)

Running flutter doctor...
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel master, v1.10.15-pre.387, on Microsoft Windows [Version 10.0.18362.449], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 29.0.0)
[√] Android Studio (version 3.5)
[√] VS Code (version 1.39.2)
[!] Connected device
    ! No devices available

! Doctor found issues in 1 category.

C:\git\dart_console>
```

### PR DESCRIPTION
**Pull request by christopherfujino**
Fixes https://github.com/flutter/flutter/issues/44118

### CODE REVIEW

**Comment by andrewkolos** [comment]
Will try to review this today in the afternoon (PDT)

File: packages/flutter_tools/lib/src/commands/channel.dart [hunk 1]
```diff
@@ -17,6 +19,12 @@ class ChannelCommand extends FlutterCommand {
       help: 'Include all the available branches (including local branches) when listing channels.',
       hide: !verboseHelp,
     );
+    argParser.addFlag(
+      'cache-artifacts',
+      help: 'After switching channels, download all required binary artifacts. '
+            'This is the equivalent of running "flutter precache" with the "--all" flag.',
```
Comment (andrewkolos) [inline packages/flutter_tools/lib/src/commands/channel.dart:25]: Possible typo; I don't think `--all` is a real option.

```suggestion
            'This is the equivalent of running "flutter precache" with the "--all-platforms" flag.',
```

**Review by andrewkolos — CHANGES_REQUESTED** [review]

File: packages/flutter_tools/lib/src/commands/channel.dart (hunk 1 above)
Comment (andrewkolos) [reply packages/flutter_tools/lib/src/commands/channel.dart:25]: Ah never mind I see you added an alias

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/commands/channel.dart:25]: It's now an alias: https://github.com/flutter/flutter/pull/118129/files#diff-e95a30630956e29a2d852d06aeccedf10ab830951e95bd6f42c9624dd7baf5a2R29

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/commands/channel.dart]: I committed your change however

**Review by andrewkolos — APPROVED** [review]
LGTM. FYI I renamed the corresponding issue to something I felt more accurate because it confused me at first. Feel free to revert/alter

**Comment by christopherfujino** [comment]
> LGTM. FYI I renamed the corresponding issue to something I felt more accurate because it confused me at first. Feel free to revert/alter

FYI this also solves the dart sdk being downloaded twice because invoking the flutter tool re-entrantly will cache the dart sdk.

**Pull request merged**